### PR TITLE
Improve documentation on publishing distributions

### PR DESCRIPTION
**Pull request by britter**
- Remove 'simply' as it's only simple for the author but not necessarily
  for the reader.
- Adapt Kotlin snippet to artifact method being able to process
  TaskProviders now.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/docs/src/snippets/ivy-publish/distribution/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -28,7 +28,7 @@ distributions {
 publishing {
     publications {
         create<IvyPublication>("myDistribution") {
-            artifact(tasks.distZip.get())
+            artifact(tasks.distZip)
```
Comment (britter) [inline subprojects/docs/src/snippets/ivy-publish/distribution/kotlin/build.gradle.kts:31]: TaskProvider is only supported by MavenPublication
```suggestion
            artifact(tasks.distZip.get())
```

**Comment by britter** [comment]
@bot-gradle test this

**Review by ljacomet — APPROVED** [review]

**Comment by britter** [comment]
@ljacomet should this go to master or some other branch in order to be part of 7.0 RC1?

**Comment by ljacomet** [comment]
@britter `master` was the right target. Thanks for the fix.

**Pull request merged**